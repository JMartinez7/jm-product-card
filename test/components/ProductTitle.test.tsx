import * as React from 'react';
// import * as ReactDOM from 'react-dom';
import renderer from 'react-test-renderer';
import { ProductTitle, ProductCard } from '../../src/components';
import { product1 } from '../data/products';

describe('<ProductTitle /> tests', () => {

  test('should show title in component', () => {
    
    const wrapper = renderer.create(
      <ProductTitle title="Custom Product" />
    )

    expect(wrapper.toJSON()).toMatchSnapshot();
  });

  test('should show productName in title', () => {

    const wrapper = renderer.create(
      <ProductCard product={product1}>
        {
          () => (
            <ProductTitle />
          )

        }
      </ProductCard>
    )

    expect(wrapper.toJSON()).toMatchSnapshot();

  });
});
