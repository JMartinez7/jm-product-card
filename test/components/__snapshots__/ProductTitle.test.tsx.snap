// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ProductTitle /> tests should show productName in title 1`] = `
<div
  className="productCard undefined"
>
  <span
    className="productDescription undefined"
  >
    Coffe Mug - without image
  </span>
</div>
`;

exports[`<ProductTitle /> tests should show title in component 1`] = `
<span
  className="productDescription undefined"
>
  Custom Product
</span>
`;
